File: javascript/darts/darts.js
function solve(x, y) {
    let scores = 0;

    if (isInsideInnerCircle(x, y)) {
        scores = 10;
    } else if (isInsideMiddleCircle(x, y)) {
        scores = 5;
    } else if (isInsideOuterCircle(x, y)) {
        scores = 1;
    }

    return scores;
};

function isInsideInnerCircle(x, y) {
    let isInsideInner = isInsideCircle(x, y, 1);

    return isInsideInner;
};

function isInsideMiddleCircle(x, y) {
    let isInsideMiddle = isInsideCircle(x, y, 5);

    return isInsideMiddle;
};

function isInsideOuterCircle(x, y) {
    let isInsideOuter = isInsideCircle(x, y, 10);

    return isInsideOuter;
};

function isInsideCircle(x, y, radius) {
    let distance = Math.sqrt(x * x + y * y);
    let isInside = distance <= radius;

    return isInside;
};

export { solve };

File: csharp/darts/Darts.cs
using System;

public static class Darts
{
    enum CircleScore
    {
        Inner = 10,
        Middle = 5,
        Outer = 1
    }

    enum CircleRadius
    {
        Inner = 1,
        Middle = 5,
        Outer = 10
    }

    public static int Score(double x, double y)
    {
       int scores = 0;
       double distance = Math.Sqrt(x * x + y * y);

       if (distance <= (int) CircleRadius.Inner)
       {
           scores += (int) CircleScore.Inner;
       }
       else if (distance <= (int) CircleRadius.Middle)
       {
           scores += (int) CircleScore.Middle;
       }
       else if (distance <= (int) CircleRadius.Outer)
       {
           scores += (int) CircleScore.Outer;
       }

       return scores;
    }
}
